### DISCUSSION TITLE
Generic template for Lightning + Hydra

### DISCUSSION THREAD

**Discussion by ashleve** [Lightning-AI/pytorch-lightning#5571] [Idea pool]
Recently I've been trying to develop a very general **PyTorch Lightning** + **Hydra** template. 

The idea is to extend **deep-learning-project-template** (https://github.com/PyTorchLightning/deep-learning-project-template), by just adding some Hydra config structure and simple example, which initializes modules from config parts in the most generic and obvious way.

I've been wondering if it would be a good idea to develop an official template of this sort and add it to Lightning ecosystem as an alternative for deep-learning-project-template?
Currently it seems like Hydra is still a little rough around the edges and it's not obvious how config should be structured for Lightning, but still such a template could be a useful addition in the future.

Here is a link to my version:
https://github.com/hobogalaxy/lightning-hydra-template

**Comment by carmocca** [comment]
Hi! There are a few projects already, for example:

https://github.com/romesco/hydra-lightning cc: @romesco
https://github.com/tchaton/lightning-hydra-seed cc: @tchaton

There is also a Slack channel named `#configs-hydra` for discussions. Feel free to visit!

**Reply by romesco** [reply]
Hey @hobogalaxy I'll be happy to checkout your thoughts about the right way to organize things and incorporate them into our effort! Ping me on the hydra-chat or on the lightning slack channel @carmocca mentioned!

I've been pushing for some paper deadlines recently so progress is slightly stagnated, but I'm excited to push forward soon! I'm excited to get a really nice workflow going, even if it requires some more work on the hydra end!

**Reply by addisonklinke** [reply]
Thank you all for sharing the resources! I would update the list to include [hydra-torch](https://github.com/pytorch/hydra-torch) where @romesco is a top contributor. In particular there's a detailed MNIST [tutorial](https://github.com/pytorch/hydra-torch/blob/master/examples/mnist_00.md) there
